### ISSUE TITLE
Everything shows up correctly on DOM but still doesn't pass final test.

### ISSUE THREAD

**Issue by calebsjames** [nss-day-cohort-46/How-to-Ask-for-Help#59]
**_Ticket MUST be completed before getting help from an instructor_**

#### Have you searched the other issue tickets? If not, do that first. 
- [x] Yes
- If No, stop and search.

####  REQUIRED Add Tags to Issue Ticket and Assign to Project Board
 

- [x] Added label to specific project/assignment that you need help with
- [x] Added to `c46 Help-Thread` project board

#### REQUIRED Describe your issue:
>>The NPM test says it can't read property 'map' of undefined but I am getting no issues while going through with a debugger and everything is showing up correctly.

#### REQUIRED List any error messages that appear:
TypeError: Cannot read property 'map' of undefined

#### REQUIRED What have you googled? **_(You must have at least 3 links related to your issue)_**

I have no idea what to google to ask why something isn't working when it is. 

#### REQUIRED: What have you tried? Tell us everything you have tried. 
>> changing properties to make sure everything is actually getting passed through. Debugging and watching every value used. 

#### REQUIRED: Copy/paste link TO THE FILE on your branch that you are having issues with

It won't let me add this as a repo. Another thing I could use help with apparently.

#### Code Snippet

```
const contentTarget = document.querySelector(".potteryList")

export const PotteryList = (allPottery) => {
    
    contentTarget.innerHTML = `
    ${allPottery.map(ap => `<section class="pottery" id="pottery--${ap.id}">
        <h2 class="pottery__shape">${ap.shape}</h2>
        <div class="pottery__properties">
            Item weighs ${ap.weight} grams and is ${ap.height} cm in height
        </div>
        <div class="pottery__price">
            Price is $${ap.price}
        </div>
    </section>
    `

    ).join("")}`
}

```

**Comment by ajawashington** [comment]
@calebsjames please refer to this issue ticket #58

**Comment by Scott47** [comment]
@calebsjames May I see where you are calling `PotterList()`?

**Comment by Scott47** [comment]
Oh. @calebsjames . What @ajawashington said.

**Comment by calebsjames** [comment]
I literally spent hours trying to figure out why it wasn't mapping correctly when that wasn't even remotely the issue nor was it the issue mentioned on ticket 56. Half my day wasted trying to fix things that were not the problem.

**Comment by calebsjames** [comment]
If anyone sees this, you are not allowed to pass an argument through PotteryList. I had to give it allPottery inside the PotteryList function. and that's what fixed mine. @ajawashington @Scott47

**Comment by ajawashington** [comment]
@calebsjames yes that's why I sent you to that ticket. Before he closes it he has a code snippet of how he fixed his issue. Which is the exact way you fixed your solution.

**Comment by calebsjames** [comment]
> @calebsjames yes that's why I sent you to that ticket. Before he closes it he has a code snippet of how he fixed his issue. Which is the exact way you fixed your solution.

Thanks @ajawashington I saw his code and made sure that PotteryList() only returned a single string of HTML
 and then on main.js, captured that return value, stored it in a variable, and set innerHTML to that variable which was what he said his solution was. That wasn't what fixed my problem though. My specific problem was that I was passing my objects as a parameter into PotteryList. 

so my code looked like this:
MAIN.JS
`const allPottery = usePottery()
let html = PotteryList(allPottery)
const contentTarget = document.querySelector(".potteryList")
contentTarget.innerHTML = html`

POTTERYLIST.JS
`export const PotteryList = (allPottery) => {

    return allPottery.map(ap => {
        return`
        <section class="pottery" id="pottery--${ap.id}">
        <h2 class="pottery__shape">${ap.shape}</h2>
        <div class="pottery__properties">
            Item weighs ${ap.weight} grams and is ${ap.height} cm in height
        </div>
        <div class="pottery__price">
            Price is $${ap.price}
        </div>
    </section>
    `
}).join("")
}`

and what fixed was putting allPottery = usePottery() within PotteryList which seems different than Dylans problem. Maybe it's not and I don't get it? Here is the code that worked for me:

MAIN.JS
`let html = PotteryList()
const contentTarget = document.querySelector(".potteryList")
contentTarget.innerHTML = html`

POTTERYLIST.JS
`export const PotteryList = () => {
    const allPottery = usePottery()
    return allPottery.map(ap => {
        return`
        <section class="pottery" id="pottery--${ap.id}">
        <h2 class="pottery__shape">${ap.shape}</h2>
        <div class="pottery__properties">
            Item weighs ${ap.weight} grams and is ${ap.height} cm in height
        </div>
        <div class="pottery__price">
            Price is $${ap.price}
        </div>
    </section>
    `
}).join("")
}`

If this was causing the same problem (not returning a single string) I'd love to get some time with an instructor to talk about it some more. Both examples above "technically" worked but the first didn't pass the test so my take away was that it didn't like me passing arguments through PotteryList(). Maybe I read something wrong that said not to do that?  If there is a benefit to use one way over the other i'd love to learn something new :-)

**Comment by Scott47** [comment]
@calebsjames There is nothing that is wrong about how you did it except that the instructions and the test do not call for passing an argument to `PotteryList()`. In test driven development, you must code to the test because that is what is required. The issue was addressed in the ticket we referred you to. Blu and Harlin as well as a few others I interacted with today also had this issue. @hrlnsnchz provided a really great example of his solution on the ticket: https://github.com/nss-day-cohort-46/How-to-Ask-for-Help/issues/56 Thank you, btw, @hrlnsnchz !
